agent/structs/connect_ca.go

```diff
@@ -35,7 +35,7 @@ type IndexedCARoots struct {
 
 	// QueryMeta contains the meta sent via a header. We ignore for JSON
 	// so this whole structure can be returned.
-	QueryMeta `json:"-"`
+	QueryMeta `json:"-" codec:"query_meta"`
```

Comment:
Another nitpick - it doesn't matter at all currently as msgpack encodings are all internal and we never look at the raw form but should we make this snake case?

The fields with no tags will end up presumaly being encoded as they are so if you ever did unmarshall the msgpack into a map[string]interface{} or translate it directly to JSON it would end up looking like:

```
{
  "ActiveRootID": "...",
  "TrustDomain": "...".
  "query_meta": {},
}
```

Can you maybe double check that? I think if `codec` follows the `json` precedent, you can specify nothing and it will just use the default name. For example you can do `json:",omitempty"` and since the name part is blank it uses the same name as if there were no JSON tag. If you can do the something similar like `codec:""` without it defaulting to `json` then that seems better.

I'd also be tempted to add a comment after each like 

```
... `json:"-" codec:""` // codec tag to override json for msgpack encoding
```

It that's too long just `// codec tag required, see #4682` or something?